src/android/ManagerService.java

```diff
@@ -148,6 +153,10 @@ public void createAndSendEvent(JSONObject obj) {
     }
 
     public void stopServiceIfInactive() {
+        if (PendingUpload.count(PendingUpload.class) > 0 && !isNetworkAvailable) {
+            updateNotificationContent("Waiting for connection");
+        }
+
```

Comment:
Est-ce que la logique que l'on veut n'est pas plutot: @azharbeebeejaun  ?


```suggestion
        if(this.connectedPlugin != null) { return; }
        if (PendingUpload.count(PendingUpload.class) == 0) {
            Intent intent = new Intent(ManagerService.this, ManagerService.class);
            this.requestObserver.unregister();
            this.requestObserver = null;
            stopService(intent);
        }else{
             if (!isNetworkAvailable) {
                updateNotificationContent("Waiting for connection");
             }
        }
```

Reply:
Oui. @sc-nick ce code resume bien le comportement voulu, non ?